alarm-monitor.py

```diff
@@ -16,122 +17,167 @@
-
-def on_message(client, userdata, message):
-    time.sleep(1)
-    print("received message =",str(message.payload.decode("utf-8")))
-
-client = paho.Client()
-
-client.username_pw_set(broker_user, broker_pass)
-client.on_message=on_message
-
-print("connecting to broker ", broker_url)
-client.connect(broker_url, broker_port)
-client.loop_start()
-
-class TexecomConnectMqtt(TexecomConnect):
-    # Overload get_zone_details to publish zone information to MQTT
-    def get_zone_details(self, zone_number):
-        zone = super(TexecomConnectMqtt, self).get_zone_details(zone_number)
-        if zone.zoneType != self.ZONETYPE_UNUSED:
-            if zone.zoneType == 1:
-                HAZoneType = "door"
-            elif zone.zoneType == 8:
-                HAZoneType = "safety"
-            else:
-                HAZoneType = "motion"
-            name = str.lower((zone.text).replace(" ", "_"))
-            topicbase = str("homeassistant/binary_sensor/" + name)
-            configtopic = str(topicbase + "/config")
-            statetopic = str(topicbase + "/state")
-            message = {
-                "name": name,
-                "device_class": HAZoneType,
-                "state_topic": statetopic,
-                "payload_on": "1",
-                "payload_off": "0",
-                "unique_id": ".".join([self.panelType, name]),
-                "device": {
-                    "name": "Texecom " + self.panelType + " " + str(self.numberOfZones),
-                    "identifiers": "123456789", #TODO panel serial number?
-                    "manufacturer": "Texecom",
-                    "model": self.panelType + " " + str(self.numberOfZones)
-                }
+
+class TexecomMqtt:
+
+    log_mqtt_traffic = False
+
+    @staticmethod
+    def on_connect(client, userdata, flags, rc):
+        if len(topic_subs[0]) > 0:
+            client.subscribe(topic_root + "/alarm_control_panel/+/command/#")
+
+    @staticmethod
+    def on_message(client, userdata, message):
+        # Arm/Disarm/Reset
+        # To give a level of security its advisable to limit access to the subscribe topic on the mqtt broker
+        topic = message.topic
+        if TexecomMqtt.log_mqtt_traffic:
+            print(
+                "topic: " + topic + " received message =",
+                str(message.payload.decode("utf-8")),
+            )
+        topicbase = topic_root + "/alarm_control_panel/"
+        if len(topic) > len(topicbase):
+            idx = topic.find("/", len(topicbase))
+            if idx >= 0:
+                subtopic = topic[len(topicbase) : idx]
+                if subtopic in topic_subs:
+                    subtopicIdx = topic_subs.index(subtopic)
+                    if len(topic_areamaps) >= subtopicIdx:
+                        areamap = topic_areamaps[subtopicIdx]
+                        area_bitmap = bytes.fromhex(areamap)
+                        if message.payload.decode("utf-8") == "arm":
```

Comment:
The default message from home assistant is 
```suggestion
                        if message.payload.decode("utf-8") == "ARM_AWAY":
```